TACC/Core-Portal
# task/TV3-173, task/TV3-174: Regex validation; Dynamic `parameterSet`

## Overview
- Enables regex validation on fields in `parameterSet` if that field has `notes::validator` defined in the form of:
```
{
    "notes": {
      "validator": {
        "regex": "",
        "message": ""
      }
    }
}
```
- Dynamically creates a schema for all fields in a `parameterSet`, including `envVariables`
- Makes fields readonly if they have `inputMode == "FIXED"`

## Related

* [TV3-173](https://jira.tacc.utexas.edu/browse/TV3-173)
* [TV3-174](https://jira.tacc.utexas.edu/browse/TV3-174)

## Testing

1. In https://cep.test/core/admin/workspace/apptrayentry/, add the app `test-app`
2. Go to https://cep.test/workbench/applications/test-app and verify the validations work, etc
3. Go to https://cep.test/workbench/applications/qgis?appVersion=3.30 and confirm you can see `envVariables`, `containerArgs`, and `schedulerOptions`

## UI
Need some feedback on UI. I have two cases:

- Minimal
![Screenshot 2023-06-06 at 5 02 21 PM](https://github.com/TACC/Core-Portal/assets/20326896/d3844942-c5bd-4cca-bfc0-004c051cd7ac)

- Verbose
![Screenshot 2023-06-06 at 5 00 58 PM](https://github.com/TACC/Core-Portal/assets/20326896/ff89c213-22f5-4653-ac63-4940ed003454)

## Notes

- I will add a `isHidden: true` param to the `notes` field of the fields we don't want users to adjust for all WMA apps before deployment
